From: Joseph Myers <joseph@codesourcery.com>
Date: Sat, 7 Nov 2009 13:30:54 +0000 (+0000)
Subject: Fix M68K fallocate for -D_FILE_OFFSET_BITS=64.
X-Git-Tag: glibc-2.11~12
X-Git-Url: http://sourceware.org/git/?p=glibc-ports.git;a=commitdiff_plain;h=1789cafb5705e51f83cdc2befa74fc9932e77f3c

Fix M68K fallocate for -D_FILE_OFFSET_BITS=64.

	* sysdeps/unix/sysv/linux/m68k/bits/fcntl.h (fallocate): Fix types
	of last two arguments when -D_FILE_OFFSET_BITS=64.
---

#diff --git a/ChangeLog.m68k b/ChangeLog.m68k
#index 59be68c..6d8c26f 100644
#--- a/ChangeLog.m68k
#+++ b/ChangeLog.m68k
#@@ -1,3 +1,8 @@
#+2009-11-07  Joseph Myers  <joseph@codesourcery.com>
#+
#+	* sysdeps/unix/sysv/linux/m68k/bits/fcntl.h (fallocate): Fix types
#+	of last two arguments when -D_FILE_OFFSET_BITS=64.
#+
# 2009-10-30  Joseph Myers  <joseph@codesourcery.com>
# 
# 	[BZ #10609]
diff --git a/sysdeps/unix/sysv/linux/m68k/bits/fcntl.h b/sysdeps/unix/sysv/linux/m68k/bits/fcntl.h
index 1e396dc..3d9ef2e 100644
--- a/sysdeps/unix/sysv/linux/m68k/bits/fcntl.h
+++ b/sysdeps/unix/sysv/linux/m68k/bits/fcntl.h
@@ -239,8 +239,8 @@ extern ssize_t tee (int __fdin, int __fdout, size_t __len,
 extern int fallocate (int __fd, int __mode, __off_t __offset, __off_t __len);
 # else
 #  ifdef __REDIRECT
-extern int __REDIRECT (fallocate, (int __fd, int __mode, __off_t __offset,
-				   __off_t __len),
+extern int __REDIRECT (fallocate, (int __fd, int __mode, __off64_t __offset,
+				   __off64_t __len),
 		       fallocate64);
 #  else
 #   define fallocate fallocate64
